### ISSUE TITLE
how do I use self-hosted Typesense that is reverse proxy by Nginx?

### ISSUE THREAD

**Issue by IskandarAlex2** [typesense/typesense#1347]
I have a VPS that will provide a few services like Firestore caching. I also want to host Typesense on it and make it all use the same port 80 I wanted to use Nginx so I can make it so that Typesense will be on `/blog_search/...` and the cache server on `/fscache/...`. However, from what I see in Typesense documentation it still wants the port to the Typesense server, and by not providing it only returns 404 in the JS SDK even with simple `client.collections().create(schema)`, I did confirm the server health is ok by going to `/blog_search/health` and of course, it responded `{"ok":true}`. Typesense docker-compose itself host to `127.0.0.1:10001` which means it is completely isolated from the global network and only Nginx reverse proxy can give access to it.

I am very sorry if my English is wrong and if the question is easy to solve, I don't have much experience.

**Comment by jasonbosco** [comment]
Could you elaborate on what your question or issue is?

**Comment by jasonbosco** [comment]
From engine's perspective, Typesense is just like any other HTTP service. 

So if you lookup documentation for "nginx reverse proxy sub-path", you should find several tutorials on this topic.

**Comment by IskandarAlex2** [comment]
m... I will try to experiment with it so I'll keep it open, I'll post if I find a solution. Still need to figure out why the Typesense npm module kept returning me a 404

**Comment by IskandarAlex2** [comment]
So I found out that Typesense npm package is not using the path properly as for reverse proxy with the location I set. I set so that any invalid path returns 400, basically switching 404 to 400 including the index page. Typesense package returned 400 when I tried to ran the same request that returned 404 before which means it is using a path that did not start with `example.com/blog_search`. I already tried using Typesense postman and it did not respond with 400 at all so all I can think of is making my own module to interact with Typesense so that's my solution for now.
